astropy/tests/test_logger.py

```diff
@@ -19,9 +19,12 @@
 _showwarning = warnings.showwarning
 
 try:
-    ip = get_ipython()
+    # this name is only defined if running within ipython/jupyter
+    __IPYTHON__
 except NameError:
-    ip = None
+    _WITHIN_IPYTHON = False
+else:
+    _WITHIN_IPYTHON = True
```

Comment:
I think you can import `_WITHIN_IPYTHON` from `logger.py` instead of redefining it here?

Reply:
done !